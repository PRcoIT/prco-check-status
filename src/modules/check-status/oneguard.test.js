const rimraf = require("rimraf");
const {
  oneguardSoapResponseGood,
  oneguardSoapResponseMissing,
} = require("./__fixtures__/oneguard-envelope");
const checkStatus = require("./check-status").checkStatus;
const getOptions = require("../../utils/get-options").getOptions;

describe("prco-check-status oneguard support", () => {
  let args;

  describe("test environment", () => {
    beforeAll(() => {
      rimraf.sync(__dirname + "/__downloads__/oneguard/test/a/b/*");
      rimraf.sync(__dirname + "/__downloads__/oneguard/test/b o b/*");
      process.env.wis_credentials = "foo,bar";
      process.env.onegard_credentials = "foo,bar";
    });

    beforeEach(() => {
      fetch.resetMocks();
      args = [
        "node",
        "prco-check-status",
        "-s",
        "oneguard",
        "-e",
        "test",
        "15748,/oneguard/test/a/b/111",
      ];
    });

    describe("when passed a valid request", () => {
      describe("using paths with spaces", () => {
        it("should return fomatted report with one entry", async () => {
          const response = {
            body: oneguardSoapResponseGood,
            status: 222,
          };
          fetch.once(() => Promise.resolve(response));

          process.argv = [...args.slice(0, -1), "15748,/oneguard/test/b o b/333"];
          const options = await getOptions();
          const responses = await checkStatus(options);
          const report = responses[0].prco.finalReport;

          expect(fetch.mock.calls.length).toEqual(1);
          expect(report).toMatchSnapshot();
        });
      });

      describe("using paths without spaces", () => {
        it("should return fomatted report with one entry", async () => {
          const response = {
            body: oneguardSoapResponseGood,
            status: 222,
          };
          fetch.once(() => Promise.resolve(response));

          process.argv = [...args];
          const options = await getOptions();
          const responses = await checkStatus(options);
          const report = responses[0].prco.finalReport;

          expect(fetch.mock.calls.length).toEqual(1);
          expect(report).toMatchSnapshot();
        });
      });
    });

    describe("when passed multiple valid requests", () => {
      it("should return fomatted report with mutiple entrys", async () => {
        const response = {
          body: oneguardSoapResponseGood,
          status: 222,
        };
        fetch.once(() => Promise.resolve(response));
        fetch.once(() => Promise.resolve(response));

        process.argv = [...args, "15748,/oneguard/test/a/b/222"];
        const options = await getOptions();
        const responses = await checkStatus(options);
        const report = responses[0].prco.finalReport;

        expect(fetch.mock.calls.length).toEqual(2);
        expect(report).toMatchSnapshot();
      });
    });

    describe("when passed an invalid request", () => {
      it("should return fomatted report with one message: Status unavailable", async () => {
        const response = {
          body: oneguardSoapResponseMissing,
          status: 222,
        };
        fetch.once(() => Promise.resolve(response));

        process.argv = [...args.slice(0, -1), "838317,/oneguard/test/a/b/555"];
        const options = await getOptions();
        const responses = await checkStatus(options);
        const report = responses[0].prco.finalReport;

        expect(fetch.mock.calls.length).toEqual(1);
        expect(report).toMatchSnapshot();
      });
    });

    describe("when passed multiple invalid requests", () => {
      it("should return fomatted report with mutiple messages: Status unavailable", async () => {
        const response = {
          body: oneguardSoapResponseMissing,
          status: 222,
        };
        fetch.once(() => Promise.resolve(response));
        fetch.once(() => Promise.resolve(response));

        process.argv = [
          ...args.slice(0, -1),
          "838317,/oneguard/test/a/b/666",
          "838317,/oneguard/test/a/b/777",
        ];
        const options = await getOptions();
        const responses = await checkStatus(options);
        const report = responses[0].prco.finalReport;

        expect(fetch.mock.calls.length).toEqual(2);
        expect(report).toMatchSnapshot();
      });
    });

    describe("when encountering a server error", () => {
      it("should return fomatted report with message: Server error", async () => {
        const errorResponse = {
          body: oneguardSoapResponseGood,
          status: 500,
        };
        fetch.once(() => Promise.resolve(errorResponse));

        process.argv = [...args];
        const options = await getOptions();
        const responses = await checkStatus(options);
        const report = responses[0].prco.finalReport;

        expect(fetch.mock.calls.length).toEqual(1);
        expect(report).toMatchSnapshot();
      });
    });
  });

  describe("production environment", () => {
    beforeAll(() => {
      rimraf.sync(__dirname + "/__downloads__/oneguard/production/a/b/*");
      rimraf.sync(__dirname + "/__downloads__/oneguard/production/b o b/*");
      process.env.wis_credentials = "foo,bar";
      process.env.onegard_credentials = "foo,bar";
    });

    beforeEach(() => {
      fetch.resetMocks();
      args = [
        "node",
        "prco-check-status",
        "-s",
        "oneguard",
        "-e",
        "production",
        "197961,/oneguard/production/a/b/111",
      ];
    });

    describe("when passed a valid request", () => {
      describe("using paths with spaces", () => {
        it("should return fomatted report with one entry", async () => {
          const response = {
            body: oneguardSoapResponseGood,
            status: 222,
          };
          fetch.once(() => Promise.resolve(response));

          process.argv = [...args.slice(0, -1), "197961,/oneguard/production/b o b/333"];
          const options = await getOptions();
          const responses = await checkStatus(options);
          const report = responses[0].prco.finalReport;

          expect(fetch.mock.calls.length).toEqual(1);
          expect(report).toMatchSnapshot();
        });
      });
      describe("using paths without spaces", () => {
        it("should return fomatted report with one entry", async () => {
          const response = {
            body: oneguardSoapResponseGood,
            status: 222,
          };
          fetch.once(() => Promise.resolve(response));

          process.argv = [...args];
          const options = await getOptions();
          const responses = await checkStatus(options);
          const report = responses[0].prco.finalReport;

          expect(fetch.mock.calls.length).toEqual(1);
          expect(report).toMatchSnapshot();
        });
      });
    });

    describe("when passed multiple valid requests", () => {
      it("should return fomatted report with mutiple entrys", async () => {
        const response = {
          body: oneguardSoapResponseGood,
          status: 222,
        };
        fetch.once(() => Promise.resolve(response));
        fetch.once(() => Promise.resolve(response));

        process.argv = [...args, "197961,/oneguard/production/a/b/222"];
        const options = await getOptions();
        const responses = await checkStatus(options);
        const report = responses[0].prco.finalReport;

        expect(fetch.mock.calls.length).toEqual(2);
        expect(report).toMatchSnapshot();
      });
    });

    describe("when passed an invalid request", () => {
      it("should return fomatted report with one message: Status unavailable", async () => {
        const response = {
          body: oneguardSoapResponseMissing,
          status: 222,
        };
        fetch.once(() => Promise.resolve(response));

        process.argv = [...args.slice(0, -1), "197961,/oneguard/production/a/b/555"];
        const options = await getOptions();
        const responses = await checkStatus(options);
        const report = responses[0].prco.finalReport;

        expect(fetch.mock.calls.length).toEqual(1);
        expect(report).toMatchSnapshot();
      });
    });

    describe("when passed multiple invalid requests", () => {
      it("should return fomatted report with mutiple messages: Status unavailable", async () => {
        const response = {
          body: oneguardSoapResponseMissing,
          status: 222,
        };
        fetch.once(() => Promise.resolve(response));
        fetch.once(() => Promise.resolve(response));

        process.argv = [
          ...args.slice(0, -1),
          "197961,/oneguard/production/a/b/666",
          "196361,/oneguard/production/a/b/777",
        ];
        const options = await getOptions();
        const responses = await checkStatus(options);
        const report = responses[0].prco.finalReport;

        expect(fetch.mock.calls.length).toEqual(2);
        expect(report).toMatchSnapshot();
      });
    });

    describe("when encountering a server error", () => {
      it("should return fomatted report with message: Server error", async () => {
        const errorResponse = {
          body: oneguardSoapResponseGood,
          status: 500,
        };
        fetch.once(() => Promise.resolve(errorResponse));

        process.argv = [...args];
        const options = await getOptions();
        const responses = await checkStatus(options);
        const report = responses[0].prco.finalReport;

        expect(fetch.mock.calls.length).toEqual(1);
        expect(report).toMatchSnapshot();
      });
    });
  });
});
