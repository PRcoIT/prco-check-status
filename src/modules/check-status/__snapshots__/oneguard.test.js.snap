// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prco-check-status oneguard support production environment when encountering a server error should return fomatted report with message: Server error 1`] = `
"
prco-check-status
server: oneguard
environment: production

---
Server error for request id: 197961


Finished

"
`;

exports[`prco-check-status oneguard support production environment when passed a valid request using paths with spaces should return fomatted report with one entry 1`] = `
"
prco-check-status
server: oneguard
environment: production

---
requestId: 197961
status: closed
state: 7
message: Inspection Request Complete
report: https://test.oneguardinspections.com/components/com_rstickets/files/2020/05/04/PRCOTEST7/PRCO TEST-0000000007_PDF_Report.pdf

Finished

"
`;

exports[`prco-check-status oneguard support production environment when passed a valid request using paths without spaces should return fomatted report with one entry 1`] = `
"
prco-check-status
server: oneguard
environment: production

---
requestId: 197961
status: closed
state: 7
message: Inspection Request Complete
report: https://test.oneguardinspections.com/components/com_rstickets/files/2020/05/04/PRCOTEST7/PRCO TEST-0000000007_PDF_Report.pdf

Finished

"
`;

exports[`prco-check-status oneguard support production environment when passed an invalid request should return fomatted report with one message: Status unavailable 1`] = `
"
prco-check-status
server: oneguard
environment: production

---
Status unavailable for: 197961



Finished

"
`;

exports[`prco-check-status oneguard support production environment when passed multiple invalid requests should return fomatted report with mutiple messages: Status unavailable 1`] = `
"
prco-check-status
server: oneguard
environment: production

---
Status unavailable for: 197961



---
Status unavailable for: 196361



Finished

"
`;

exports[`prco-check-status oneguard support production environment when passed multiple valid requests should return fomatted report with mutiple entrys 1`] = `
"
prco-check-status
server: oneguard
environment: production

---
requestId: 197961
status: closed
state: 7
message: Inspection Request Complete
report: https://test.oneguardinspections.com/components/com_rstickets/files/2020/05/04/PRCOTEST7/PRCO TEST-0000000007_PDF_Report.pdf

---
requestId: 197961
status: closed
state: 7
message: Inspection Request Complete
report: https://test.oneguardinspections.com/components/com_rstickets/files/2020/05/04/PRCOTEST7/PRCO TEST-0000000007_PDF_Report.pdf

Finished

"
`;

exports[`prco-check-status oneguard support test environment when encountering a server error should return fomatted report with message: Server error 1`] = `
"
prco-check-status
server: oneguard
environment: test

---
Server error for request id: 15748


Finished

"
`;

exports[`prco-check-status oneguard support test environment when passed a valid request using paths with spaces should return fomatted report with one entry 1`] = `
"
prco-check-status
server: oneguard
environment: test

---
requestId: 15748
status: closed
state: 7
message: Inspection Request Complete
report: https://test.oneguardinspections.com/components/com_rstickets/files/2020/05/04/PRCOTEST7/PRCO TEST-0000000007_PDF_Report.pdf

Finished

"
`;

exports[`prco-check-status oneguard support test environment when passed a valid request using paths without spaces should return fomatted report with one entry 1`] = `
"
prco-check-status
server: oneguard
environment: test

---
requestId: 15748
status: closed
state: 7
message: Inspection Request Complete
report: https://test.oneguardinspections.com/components/com_rstickets/files/2020/05/04/PRCOTEST7/PRCO TEST-0000000007_PDF_Report.pdf

Finished

"
`;

exports[`prco-check-status oneguard support test environment when passed an invalid request should return fomatted report with one message: Status unavailable 1`] = `
"
prco-check-status
server: oneguard
environment: test

---
Status unavailable for: 838317



Finished

"
`;

exports[`prco-check-status oneguard support test environment when passed multiple invalid requests should return fomatted report with mutiple messages: Status unavailable 1`] = `
"
prco-check-status
server: oneguard
environment: test

---
Status unavailable for: 838317



---
Status unavailable for: 838317



Finished

"
`;

exports[`prco-check-status oneguard support test environment when passed multiple valid requests should return fomatted report with mutiple entrys 1`] = `
"
prco-check-status
server: oneguard
environment: test

---
requestId: 15748
status: closed
state: 7
message: Inspection Request Complete
report: https://test.oneguardinspections.com/components/com_rstickets/files/2020/05/04/PRCOTEST7/PRCO TEST-0000000007_PDF_Report.pdf

---
requestId: 15748
status: closed
state: 7
message: Inspection Request Complete
report: https://test.oneguardinspections.com/components/com_rstickets/files/2020/05/04/PRCOTEST7/PRCO TEST-0000000007_PDF_Report.pdf

Finished

"
`;
